python/ray/tune/tests/test_trial_runner_3.py

```diff
@@ -610,6 +611,53 @@ def testUserCheckpoint(self):
         self.assertEqual(ray.get(trials2[0].runner.get_info.remote()), 1)
         shutil.rmtree(tmpdir)
 
+    @patch("ray.tune.ray_trial_executor.TUNE_RESULT_BUFFER_MIN_TIME_S", 1)
+    @patch("ray.tune.ray_trial_executor.TUNE_RESULT_BUFFER_LENGTH", 8)
+    def testUserCheckpointBuffered(self):
+        def num_checkpoints(trial):
+            return sum(
+                item.startswith("checkpoint_")
+                for item in os.listdir(trial.logdir))
+
+        ray.init(num_cpus=3)
+        tmpdir = tempfile.mkdtemp()
```

Comment:
I think setup already creates a tmpdir?

Reply:
Ah, I copied that from `testUserCheckpoint`, but it seems you're right. Adjusted both tests accordingly.